compiler/resolution/functionResolution.cpp

```diff
@@ -1980,20 +1979,34 @@ static bool prefersCoercionToOtherNumericType(Type* actualType,
         !(f2Type == dtInt[INT_SIZE_DEFAULT] ||
           f2Type == dtUInt[INT_SIZE_DEFAULT]))
       return true;
-    // Prefer bool/enum/int/uint cast to a default-sized real over another
-    // size of real or complex.
-    if ((aBoolEnum || aT == NUMERIC_TYPE_INT_UINT) &&
-        f1Type == dtReal[FLOAT_SIZE_DEFAULT] &&
-        (f2T == NUMERIC_TYPE_REAL || f2T == NUMERIC_TYPE_COMPLEX) &&
-        f2Type != dtReal[FLOAT_SIZE_DEFAULT])
-      return true;
-    // Prefer bool/enum/int/uint cast to a default-sized complex over another
-    // size of complex.
-    if ((aBoolEnum || aT == NUMERIC_TYPE_INT_UINT) &&
-        f1Type == dtComplex[COMPLEX_SIZE_DEFAULT] &&
-        f2T == NUMERIC_TYPE_COMPLEX &&
-        f2Type != dtComplex[COMPLEX_SIZE_DEFAULT])
-      return true;
+    // For non-default-sized ints/uints...
```

Comment:
I'm wondering if we should express it in code as "prefer the corresponding real size to other real sizes" where "the corresponding real size" would be defined as:
 * 32 for numbers with width <= 32
 * 64 in all other cases

I think this works out the same because there are only two cases (ignoring for now the variety of int/uint/real/complex), really:
 * prefer real(32) for int(?n) with n <= 32
 * prefer real(64) for int(64)

I'm not so sure we want implicit conversions from bool/enum to real/complex at all (but I don't remember where the language is on that right now).

Reply:
In this section of the code, it's actually just one case currently since we're already in an "if this is an integer whose width is less than the default" conditional, so it's just the first of your two bullets in each of the above lists.  I'm still open to rewording the comment (for the sake of admitting possible new real sizes in the future I guess?), but don't think it ought to refer to the 64-bit integers since they're not being handled here (?).  Anyway, let me know how you'd like me to proceed.

> I'm not so sure we want implicit conversions from bool/enum to real/complex at all (but I don't remember where the language is on that right now).

I agree with this (on both counts), but am not clear on why you're saying it here.  Do you think I've changed the behavior in some way?  (I thought that I hadn't since I'm within an "if this is an int or uint" conditional).

[my guess is that since enums should no longer coerce to anything, they probably don't coerce to real and complex; and I'd hope the same is true for bools.  Testing, that seems to be the case]

Reply:
It's more just that this code is written to do something to prefer certain coercions from bool/enum to numeric types. So it could be simplified in this PR to remove that logic, or maybe just add a comment that it's out-of-date code covering a case that should no longer appear.

It looks like bools still coerce to `int`, at least. I think that might be reasonable to remove, FWIW, but that is another issue.

Anyway, as far as what to do for this PR, maybe just put a comment about how this code could improve (to remove logic about enum/bool->real which is not available; and possibly to make its logic match better the spec wording).

Reply:
> It's more just that this code is written to do something to prefer certain coercions from bool/enum to numeric types.

I'm sorry, I'm still not following and may be missing something.  The code you've commented on here is within an `if (is_complex_type...` conditional, and then is filtering on `is_int_type() || is_uint_type()` so I don't see how enums and bools come into play at all.  The code I _removed_ used to handle bools and enums (so maybe you're pulling a Brad and reading the diff backwards?)

> It looks like bools still coerce to int, at least. 

I would argue to preserve this, personally.  It supports nice idioms like `numMatches = + reduce (A == "target");`.

Reply:
I will rely on individual lines

Reply:
Thanks.  For some reason (probably GitHub's default view), I only look upwards from the line where a comment is added to the code, which is why I was confused about what you were referring to.

Reply:
@mppf: OK, I think I've cleaned up that bool/enum stuff with no new failures if you want to take another look.